### PR TITLE
Test if hidden collections output a document with a future date

### PR DESCRIPTION
**Pull request by ashmaroli**
Tests #5953 

/cc @jekyll/build

### CODE REVIEW

File: features/collections.feature [hunk 1]
```diff
@@ -94,6 +94,51 @@ Feature: Collections
...
+    Then I should get a zero exit status
+    And the _site directory should exist
+    And the "_site/puppies/fido.html" file should exist
+
+  Scenario: Hidden collection with document with future date
+    Given I have a _puppies directory
+    And I have the following documents under the puppies collection:
+      | title  | date       | content             |
+      | Rover  | 2007-12-31 | content for Rover.  |
+      | Fido   | 2020-12-31 | content for Fido.   |
```
Comment (crispgm) [inline features/collections.feature]: IMO, `2020` is not "future" enough.

Comment (ashmaroli) [inline features/collections.feature]: no? In theory, even `tomorrow` is future..

Comment (crispgm) [inline features/collections.feature]: @ashmaroli Right! But 3 years is not far. And this test case will be outdated.

**Comment by parkr** [comment]
@jekyllbot: merge +dev

**Pull request merged**